packages/automerge-repo-react-hooks/test/useDocument.test.tsx

```diff
@@ -64,150 +49,268 @@ describe("useDocument", () => {
     url,
     onDoc,
   }: {
-    url: AutomergeUrl | undefined
+    url: AutomergeUrl
     onDoc: (doc: ExampleDoc) => void
   }) => {
-    const [doc] = useDocument(url)
+    const [doc] = useDocument<ExampleDoc>(url)
     onDoc(doc)
-    return null
+    return <div data-testid="content">{doc.foo}</div>
   }
 
   it("should load a document", async () => {
     const { handleA, wrapper } = setup()
     const onDoc = vi.fn()
 
-    render(<Component url={handleA.url} onDoc={onDoc} />, { wrapper })
-    await waitFor(() => expect(onDoc).toHaveBeenLastCalledWith({ foo: "A" }))
-  })
+    render(
+      <Suspense fallback={<div data-testid="loading">Loading...</div>}>
+        <Component url={handleA.url} onDoc={onDoc} />
+      </Suspense>,
+      { wrapper }
+    )
 
-  it("should immediately return a document if it has already been loaded", async () => {
-    const { handleA, wrapper } = setup()
-    const onDoc = vi.fn()
+    // First we should see the loading state
+    expect(screen.getByTestId("loading")).toBeInTheDocument()
```

Comment:
We expect to see a loading state even if we're dealing with a locally-created document?